dump_manager_bmc.cpp

```diff
@@ -177,7 +177,18 @@ void Manager::checkAndCreateCoreDump()
             log<level::INFO>(
                 fmt::format("Core file found, files size {}", files.size())
                     .c_str());
-            captureDump(Type::ApplicationCored, files);
+            try
+            {
+                captureDump(Type::ApplicationCored, files);
```

Comment:
Couple of issue here:
Based on the defect journal logs "Fri Oct  7 19:31:11 2022 ever13bmc phosphor-dump-manager: Core file found, files size 64"
There are 64 core's available , which includes dump manger/hw diags core.

ApplicationCored type only supports to process  only one core and  not supported for multiple dump files. This type is not intended  for this.

 checkAndCreateCoreDump should be  light weight user initiated dump which copies all the available cores from core folder and deletes the files or should initiate each core file bases application core dump , which may be little slower.

Not seeing these changes in  upstream. can you please include upstream  reveiw link related to this feature enablement .  so that  we can do more discussion there. 

Also need to look systemd coredump  auto delete policy , like anything we need to optimize here. on core dump directory max size.

Reply:
Hi @ojayanth , 

Please find below the Gerrit link for the original feature enhancement.

https://gerrit.openbmc.org/c/openbmc/phosphor-debug-collector/+/54041